Documentation/guides/BindingJavaSource.md

```diff
@@ -0,0 +1,97 @@
+---
+id:
+title: "Binding Java Source"
+dateupdated: 2021-05-21
+---
+
+# Overview
+
+Bindings can be done on not only .jar and .aar files but also raw
+.java code. This will allow developers of bindings or applications
+to write custom API's for underlying bindings and expose them
+easily to the developer.
+
+Applications already had the ability to include `AndroidJavaSource`
+items in them, these would then be compiled into the final `classes.dex`.
+However this custom java source code was not "bound" and if users wanted
+to call these classes they would need to do that manually.
+
+The `AndroidJavaSource` Item Group now supports a `Bind` attribute. This
+will instruct the build system to not only compile the code but also
+produce a C# API for it.
+
+```xml
+<ItemGroup>
+    <AndroidJavaSource Include="MyClass.java" Bind="True" />
+</ItemGroup>
+```
+
+The `Bind` attribute will default to `true` when in binding projects.
```

Comment:
In a .NET 6 context, what constitutes a "binding project"?  "Binding project" is a noun without a future, and leaves things confusing in a .NET 6 context.

It would likely be clearer to say:

```markdown
The `%(Bind)` attribute will default to `true` in Legacy Java Binding Projects,
and in .NET 6 projects in which the `$(OutputType)` MSBuild property is `Library`.
[TODO: verify]
In Legacy Xamarin.Android Library and Application projects, `%(Bind)` will default to False,
as well as in .NET 6 projects in which the `$(OutputType)` MSBuild property is `Exe`.
```

Reply:
…though see my [followup remark](https://github.com/xamarin/xamarin-android/pull/5926/files#r638287466), in which I kinda changed my mind.

Semantics are hard. :-/